src/debug/PythonDebugProvider.ts

```diff
@@ -39,7 +39,7 @@ async function getPythonCommand(host: string, port: number): Promise<string> {
 
```

Comment:
Related to discussion in https://github.com/Azure/azure-functions-python-worker/issues/607 with @mhoeger, one of the scenarios mentioned by users is as follows:
1. Create a simple python project with an http trigger
2. Make sure "AzureWebJobsStorage" is set to "UseDevelopmentStorage=true"
3. Make sure the local emulator is _not_ running and start debugging

With your change, it works. Without your change, I get "The operation has timed out". This scenario isn't related to "continuous debugging" - so can you explain why your change helps?

Reply:
Hi @ejizba, this is because in the vscode, when hitting F5, it does not start the Python azure_functions_worker directly, but starts the debugpy Python module provided by ms.python extension.

The debugpy is used for hosting a Python debug server for the Python code (**debugee**). This debugpy accepts a `--wait-for-client` flag. When this flag is set to true, it will not start the Python azure_functions_worker until the vscode debugger connects to it.

From my understanding, the vscode-azurefunctions extension has a [problem matcher](https://github.com/microsoft/vscode-azurefunctions/blob/main/package.json) **Worker process started and initialized|Host lock lease acquired by instance ID**, which only connects the vscode debugger to debugee when after the Python worker starts. This creates a racing condition.

Let me know if you have any questions.

Reply:
Right so this is my understanding of the current ordering with "wait-for-client" flag:
1. Start func cli
2. Func cli starts debugpy
3. debugpy waits for client to attach once "Worker process is started and initialized" is printed by func cli
4. debugpy starts the worker

So my question is - how does "Worker process is started and initialized" _ever_ get printed if debugpy waits until step 4 to start the worker? And why does the local emulator affect that?

Hmm now that I think about it - is it because "host lock lease" is what's triggering step 3 (since the problem matcher has two options) and that won't be printed if the emulator isn't running?